src/api/alchemy.ts

```diff
@@ -507,9 +525,395 @@ export class Alchemy {
   }
 
   /**
-   * Creates an AlchemyProvider instance. Only one provider is created per
+   * Returns the balance of a given address as of the provided block.
+   *
+   * @param addressOrName The address or name of the account to get the balance for.
+   * @param blockTag The optional block number or hash to get the balance for.
+   *   Defaults to 'latest' if unspecified.
+   * @public
+   */
+  async getBalance(
+    addressOrName: string | Promise<string>,
+    blockTag?: BlockTag | Promise<BlockTag>
+  ): Promise<BigNumber> {
+    const provider = await this.getProvider();
+    return provider.getBalance(addressOrName, blockTag);
+  }
```

Comment:
If you don't like writing these repetitive methods that pass through to the provider, there's an option to do tricks with a dynamic proxy to save writing. But this way is good too because it's straightforward.